bindings/java/src/org/sleuthkit/datamodel/OsAccountManager.java

```diff
@@ -55,121 +54,123 @@
 	}
 
 	/**
-	 * Creates an OS account with given unique id or given realm and login name.
-	 * If an account already exists with the given id or realm/login, then the
+	 * Creates an OS account with given unique id and given realm id.
+	 * If an account already exists with the given id, then the
 	 * existing OS account is returned.
 	 *
 	 * @param uniqueAccountId Account sid/uid.
-	 * @param loginName       Login name.
-	 * @param realmName       Realm within which the accountId or login name is
-	 *                        unique.
-	 * @param host            Host for the realm, may be null.
+	 * @param realmId       Realm id.
+	 * 
+	 * @return OsAccount.
+	 *
+	 * @throws TskCoreException If there is an error in creating the OSAccount.
+	 *
+	 */
+	OsAccount createOsAccount(String uniqueAccountId,  long realmId) throws TskCoreException {
+
+		// ensure unique id is provided
+		if (Strings.isNullOrEmpty(uniqueAccountId)) {
+			throw new IllegalArgumentException("Cannot create OS account with null uniqueId.");
+		}
+		
+		try (CaseDbConnection connection = this.db.getConnection();) {
+
+			OsAccountRealm realm = db.getOsAccountRealmManager().getRealm(realmId, connection);
+				
+			// try to create account
+			try {
+				return createOsAccount(uniqueAccountId, null, realm, OsAccount.OsAccountStatus.UNKNOWN, connection);
+			} catch (SQLException ex) {
+
+				// Create may fail if an OsAccount already exists. 
+				Optional<OsAccount> osAccount = this.getOsAccountByUniqueId(uniqueAccountId, realm);
+				if (osAccount.isPresent()) {
+					return osAccount.get();
+				}
+
+				// create failed for some other reason, throw an exception
+				throw new TskCoreException(String.format("Error creating OsAccount with uniqueAccountId = %s in realm id = %d", uniqueAccountId, realmId), ex);
+			}
+		} 
+	}
+
+	
+	/**
+	 * Creates an OS account with Windows-specific data. 
+	 * If an account already exists with the given id or realm/login, then the
+	 * existing OS account is returned.  
+	 *
+	 * @param sid           Account sid/uid, can be null if loginName is supplied. 
+	 * @param loginName     Login name, can be null if sid is supplied. 
+	 * @param realmName     Realm within which the accountId or login name is
+	 *                      unique. Can be null if sid is supplied. 
+	 * @param referringHost Host referring the account.
+	 * @param realmScope    Realm scope.
 	 *
 	 * @return OsAccount.
 	 *
 	 * @throws TskCoreException If there is an error in creating the OSAccount.
 	 *
 	 */
-	public OsAccount createOsAccount(String uniqueAccountId, String loginName, String realmName, Host host) throws TskCoreException {
+	public OsAccount createWindowsAccount(String sid, String loginName, String realmName, Host referringHost, OsAccountRealm.RealmScope realmScope) throws TskCoreException {
```

Comment:
@raman-bt can you please add IllegalArgumentException to the throws part in the documentation? 
There are prob other methods too, but the users of the api will not know without looking at code.